src/google/protobuf/port_def.inc

```diff
@@ -413,6 +413,26 @@
 #define PROTOBUF_ATTRIBUTE_REINITIALIZES
 #endif
 
+#ifdef PROTOBUF_NO_RETURN
+#error PROTOBUF_NO_RETURN was previously defined
+#endif
+#ifdef __cpp_has_attribute
+#if __cpp_has_attribute(noreturn)
+#define PROTOBUF_NO_RETURN [[noreturn]]
+#else
+#define PROTOBUF_NO_RETURN
+#endif
+#else
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
here you are probably want to use PROTOBUF_GNUC_MIN(x, y) with whatever version this got added in